connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/a2a/client/inbound/webhook/A2aClientWebhookExecutable.java

```diff
@@ -0,0 +1,177 @@
+          elementType = BpmnType.INTERMEDIATE_CATCH_EVENT,
+          templateIdOverride = "io.camunda.connectors.agenticai.a2a.client.webhook.intermediate.v0",
+          templateNameOverride =
+              "A2A Client Webhook Intermediate Catch Event Connector (early access)"),
+      @ConnectorElementType(
+          appliesTo = BpmnType.RECEIVE_TASK,
+          elementType = BpmnType.RECEIVE_TASK,
+          templateIdOverride = "io.camunda.connectors.agenticai.a2a.client.webhook.receive.v0",
+          templateNameOverride = "A2A Client Webhook Receive Task Connector (early access)"),
+    })
+@InboundConnector(name = "A2A Webhook Connector", type = "io.camunda.agenticai:a2aclient:webhook:0")
+public class A2aClientWebhookExecutable implements WebhookConnectorExecutable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(A2aClientWebhookExecutable.class);
+
+  private final A2aSdkObjectConverter a2aSdkObjectConverter;
+  private final ObjectMapper objectMapper;
+
+  private A2aWebhookProperties props;
+  private WebhookAuthorizationHandler<?> authChecker;
+  private InboundConnectorContext context;
+
+  public A2aClientWebhookExecutable(
+      A2aSdkObjectConverter a2aSdkObjectConverter, ObjectMapper objectMapper) {
+    this.a2aSdkObjectConverter = a2aSdkObjectConverter;
+    this.objectMapper = objectMapper;
+  }
+
+  @Override
+  public void activate(InboundConnectorContext context) {
+    this.context = context;
+    var wrappedProps = context.bindProperties(A2aWebhookPropertiesWrapper.class);
+    props = new A2aWebhookProperties(wrappedProps);
+    authChecker = WebhookAuthorizationHandler.getHandlerForAuth(props.auth());
+    context.reportHealth(Health.up());
+  }
+
+  @Override
+  public WebhookResult triggerWebhook(WebhookProcessingPayload payload) {
+    LOGGER.debug("Triggered A2A webhook with context {}", props.context());
+
+    verifySignature(payload);
+
+    final var authResult = authChecker.checkAuthorization(payload);
+    if (authResult instanceof Failure failureResult) {
+      throw failureResult.toException();
+    }
+
+    final var mappedRequest = mapRequest(payload);
+    return new A2aWebhookResult(mappedRequest);
+  }
+
+  private MappedHttpRequest mapRequest(WebhookProcessingPayload payload) {
+    try {
+      Task task = objectMapper.readValue(payload.rawBody(), Task.TYPE_REFERENCE);
+      A2aTask a2aTask = a2aSdkObjectConverter.convert(task);
+      return new MappedHttpRequest(a2aTask, payload.headers(), payload.params());
+    } catch (IOException e) {
+      this.context.log(
+          activity ->
+              activity
+                  .withSeverity(Severity.ERROR)
+                  .withTag("a2a-webhook-request")
+                  .withMessage("Error deserializing A2A Webhook payload: " + e.getMessage()));
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void verifySignature(WebhookProcessingPayload payload) {
```

Comment:
In a separate PR I will extract the verification logic to a dedicated class in the HTTP webhook connector and call it from here.